### PR TITLE
Address remaining cppwinrt 2 template issues, add desktop app template

### PR DESCRIPTION
**Pull request by Scottj1s**

### CODE REVIEW

File: src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/ConsoleApplication/pch.h [hunk 1]
```diff
@@ -1,2 +1,4 @@
 ﻿#pragma once
+#include <winrt/coroutine.h>
 #include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Foundation.Collections.h>
```
Comment (kennykerr) [inline src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/ConsoleApplication/pch.h]: Namespace headers always include their parent namespace headers.

File: src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/pch.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+#include <windows.h>
+#ifdef GetCurrentTime
+#undef GetCurrentTime
+#endif
+#include <winrt/coroutine.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.System.h>
+#include <winrt/Windows.UI.Xaml.h>
```
Comment (kennykerr) [inline src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/pch.h]: Same here.

File: src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/resource.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
+//{{NO_DEPENDENCIES}}
```
Comment (kennykerr) [inline src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/resource.h]: Why do we need this?

File: src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/ConsoleApplication/pch.h (hunk 1 above)
Comment (Scottj1s) [reply src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/ConsoleApplication/pch.h]: yes, but I like the explicit calling out so users can selectively remove one at a time

File: src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/pch.h (hunk 2 above)
Comment (Scottj1s) [reply src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/pch.h]: by design

File: src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/resource.h (hunk 3 above)
Comment (Scottj1s) [reply src/package/cppwinrt/vsix/ProjectTemplates/VC/Windows Desktop/WindowsApplication/resource.h]: RC file

**Pull request merged**